extensions/resource-deployment/src/ui/resourceTypeWizard.ts

```diff
@@ -35,36 +40,29 @@ export class ResourceTypeWizard {
 		this._useGenerateScriptButton = value;
 	}
 
-	//TODO: eventually only resourceType will be passed. For now, we are passing both the resourceType and provider
 	constructor(
 		public resourceType: ResourceType,
-		public provider: DeploymentProvider,
 		public _kubeService: IKubeService,
 		public azdataService: IAzdataService,
 		public notebookService: INotebookService,
 		public toolsService: IToolsService,
-		public platformService: IPlatformService) {
-		this.wizardObject = azdata.window.createWizard(resourceType.displayName, resourceType.name, 'wide');
-		this.model = this.getResourceProviderModel()!;
-	}
-
+		public platformService: IPlatformService,
+		public resourceTypeService: ResourceTypeService) {
 
-	public getResourceProviderModel(): ResourceTypeModel | undefined {
-		if (instanceOfWizardDeploymentProvider(this.provider)) {
-			return new DeployClusterWizardModel(this.provider, this);
-		} else if (instanceOfAzureSQLVMDeploymentProvider(this.provider)) {
-			return new DeployAzureSQLVMWizardModel(this.provider, this);
-		} else if (instanceOfNotebookWizardDeploymentProvider(this.provider)) {
-			return new NotebookWizardModel(this.provider, this);
-		} else if (instanceOfAzureSQLDBDeploymentProvider(this.provider)) {
-			return new DeployAzureSQLDBWizardModel(this.provider, this);
+		if (resourceType.options) {
+			this.provider = this.resourceType.getProvider(resourceType.options.map(option => { return { option: option.name, value: option.values[0].name }; }))!;
```

Comment:
Why is the value being set to just the first one all the time? The logic is supposed to be the currently selected options so it doesn't really make sense to me to just pick the first here unless I'm missing something.

Reply:
So in the toolsAndEulaPage, we are always getting the first value for all the dropdowns as the default value. I am in a way replicating the logic over here.

Reply:
Where are we updating the provider though? If a resource type has different providers then we need to make sure we update that if they choose a different option right?

Reply:
That is happening in the tools and Eula page

Reply:
But we have things that are created based off of the provider here - such as the model. Is that being correctly updated?

Reply:
Yup, the method setModel is called once initially to get the first model and then it is called from toolsAndEulaPage whenever we change the model.

Reply:
setModel? I don't see that function anywhere. Do you mean this call here to update the provider? https://github.com/microsoft/azuredatastudio/pull/13182/files#diff-6d9beaebb1d2b084c5a260e0fff4deebef72efaeb446465eb7c698793f27cf80R199

If that's the case then please add some comments indicating that these values are updated by the pages later on since it's not obvious that's what's happening.

And in the future this is where a "View Model" approach helps. Instead of having properties such as the current provider kept by the wizard class itself you use a model for the entire wizard itself that contains things like the current provider. Makes it easier to pass around and keeps common logic like that separated out from specific implementations.